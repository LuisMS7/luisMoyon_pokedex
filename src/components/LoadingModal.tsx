import React from "react";
import { View, Modal, ActivityIndicator, StyleSheet } from "react-native";

type LoadingModalProps = {
	visible: boolean;
};

const LoadingModal = ({ visible }: LoadingModalProps) => {
	return (
		<Modal visible={visible} transparent animationType="fade">
			<View style={styles.modalContainer}>
				<ActivityIndicator size="large" color="blue" />
			</View>
		</Modal>
	);
};

const styles = StyleSheet.create({
	modalContainer: {
		flex: 1,
		alignItems: "center",
		justifyContent: "center",
		backgroundColor: "rgba(0, 0, 0, 0.5)",
	},
});

export default LoadingModal;
