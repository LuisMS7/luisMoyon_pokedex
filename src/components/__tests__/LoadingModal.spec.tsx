import React from "react";
import { render, screen } from "@testing-library/react-native";
import LoadingModal from "@src/components/LoadingModal";

describe("LoadingModal", () => {
	it("should render when visible is true", () => {
		render(<LoadingModal visible={true} />);
		expect(screen.toJSON()).toMatchSnapshot();
	});
	it("should not render when visible is false", () => {
		render(<LoadingModal visible={false} />);
		expect(screen.toJSON()).toMatchSnapshot();
	});
});
