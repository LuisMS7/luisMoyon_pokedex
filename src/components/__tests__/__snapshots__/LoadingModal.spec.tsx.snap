// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingModal should not render when visible is false 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
/>
`;

exports[`LoadingModal should render when visible is true 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      color="blue"
      size="large"
    />
  </View>
</Modal>
`;
